pmd-java/src/test/java/net/sourceforge/pmd/lang/java/symbols/internal/TypeAnnotTestUtil.java

```diff
@@ -84,13 +84,13 @@ public static <A extends Annotation> A createAnnotationInstance(Class<A> annotat
     @SuppressWarnings("unchecked")
     public static <A extends Annotation> A createAnnotationInstance(Class<A> annotationClass, Map<String, Object> attributes) {
         return (A) Proxy.newProxyInstance(annotationClass.getClassLoader(), new Class[] { annotationClass }, (proxy, method, args) -> {
-            if (method.getName().equals("annotationType") && args == null) {
```

Comment:
if the rules and or the test matter we should apply them consequently not partially only on production.

Some could argue test are considered important, so they should receive the same same quality norms, as well.